zerver/lib/actions.py

```diff
@@ -2231,6 +2233,16 @@ def do_set_user_display_setting(user_profile, setting_name, setting_value):
              'setting': setting_value}
     send_event(event, [user_profile.id])
 
+    payload = dict(email=user_profile.email,
+                   user_id=user_profile.id,
+                   timezone=user_profile.timezone)
+
+    send_event(dict(type='realm_user', op='update', person=payload),
+               active_user_ids(user_profile.realm))
+    if user_profile.is_bot:
+        send_event(dict(type='realm_bot', op='update', bot=payload),
+                   bot_owner_userids(user_profile))
```

Comment:
We don't need an event at all for bots, and we only need the `realm_user` event if it's a timezone change, not some other display setting change.